data/actions/lib/actions.lua

```diff
@@ -39,3 +56,175 @@ function destroyItem(player, target, toPosition)
+	local tile = Tile(toPosition)
+	if not tile then
+		return false
+	end
+
+	if table.contains(ropeSpots, tile:getGround():getId()) or tile:getItemById(14435) then
+		if Tile(toPosition:moveUpstairs()):hasFlag(TILESTATE_PROTECTIONZONE) and player:isPzLocked() then
+			player:sendCancelMessage(RETURNVALUE_PLAYERISPZLOCKED)
+			return true
+		end
+		player:teleportTo(toPosition, false)
+		return true
+	elseif table.contains(holeId, target.itemid) then
+		toPosition.z = toPosition.z + 1
+		tile = Tile(toPosition)
+		if tile then
+			local thing = tile:getTopVisibleThing()
+			if thing:isPlayer() then
+				if Tile(toPosition:moveUpstairs()):hasFlag(TILESTATE_PROTECTIONZONE) and thing:isPzLocked() then
+					return false
+				end
+				return thing:teleportTo(toPosition, false)
+			end
+			if thing:isItem() and thing:getType():isMovable() then
+				return thing:moveTo(toPosition:moveUpstairs())
+			end
+		end
+		player:sendCancelMessage(RETURNVALUE_NOTPOSSIBLE)
+		return true
+	end
+	return false
+end
+
+function onUseShovel(player, item, fromPosition, target, toPosition, isHotkey)
+	local tile = Tile(toPosition)
+	if not tile then
+		return false
+	end
+
+	local ground = tile:getGround()
+	if not ground then
+		return false
+	end
+
+	local groundId = ground:getId()
+	if table.contains(holes, groundId) then
+		ground:transform(groundId + 1)
+		ground:decay()
+
+		toPosition.z = toPosition.z + 1
+		tile:relocateTo(toPosition)
+	elseif groundId == 231 then
+		local randomValue = math.random(1, 100)
+		if randomValue == 1 then
+			Game.createItem(2159, 1, toPosition)
+		elseif randomValue > 95 then
+			Game.createMonster("Scarab", toPosition)
+		end
+		toPosition:sendMagicEffect(CONST_ME_POFF)
+	else
+		return false
+	end
+
+	return true
+end
+
+function onUseScythe(player, item, fromPosition, target, toPosition, isHotkey)
+	if not table.contains({2550, 10513}, item.itemid) then
+		return false
+	end
+
+	if target.itemid == 2739 then -- wheat
+		target:transform(2737)
+		target:decay()
+		Game.createItem(2694, 1, toPosition) -- bunch of wheat
+		return true
+	end
+	if target.itemid == 5465 then -- burning sugar cane
+		target:transform(5464)
+		target:decay()
+		Game.createItem(5467, 1, toPosition) -- bunch of sugar cane
+		return true
+	end
+	return destroyItem(player, target, toPosition)
+end
+
+function onUseCrowbar(player, item, fromPosition, target, toPosition, isHotkey)
+	if not table.contains({2416, 10515}, item.itemid) then
+		return false
+	end
+
+	return destroyItem(player, target, toPosition)
+end
+
+function onUseKitchenKnife(player, item, fromPosition, target, toPosition, isHotkey)
+	if not table.contains({2566, 10511, 10515}, item.itemid) then
+		return false
+	end
+
+	if table.contains(fruits, target.itemid) and player:removeItem(6278, 1) then
+		target:remove(1)
+		player:addItem(6279, 1)
+		player:getPosition():sendMagicEffect(CONST_ME_MAGIC_GREEN)
+	else
+		return false
```

Comment:
You can return true here and false on line 229, that way you don't need the else statment.

Reply:
like this?